stock_vertical_lift/models/vertical_lift_operation_base.py

```diff
@@ -34,6 +34,15 @@ def split_other_move_lines(move, move_lines):
     return False
 
 
+def set_picking_location_dest(picking, location):
+    """Set location_dest_id on the picking and related moves."""
+    if picking.location_dest_id != location:
+        picking.location_dest_id = location.id
```

Comment:
```suggestion
        picking.location_dest_id = location
```
So you'll benefit of the type checking from Odoo (ensuring that `location` is a `stock.location` recordset)